File: cplusplus/logger.cpp
#include <iostream>

int main() {
    std::cout << R"({"timestamp":"2026-01-01T00:00:00Z","level":"INFO","service_name":"cpp-service","version":"0.0.1","trace_id":"stub","message":"hello from cpp","context":{}})" << std::endl;
    return 0;
}

File: ruby/logger.rb
require 'json'

def mask(value)
  return value.gsub(/([A-Za-z0-9._%+-]+)@([A-Za-z0-9.-]+\.[A-Za-z]{2,})/, '***@***')
end

event = {
  timestamp: Time.now.utc.iso8601,
  level: 'INFO',
  service_name: 'ruby-service',
  version: '0.0.1',
  trace_id: 'stub',
  message: 'hello from ruby',
  context: { email: mask('alice@example.com') }
}

puts JSON.generate(event)
